### PR TITLE
HOTT-2002: Adding new tests for beta search for guides

### PR DESCRIPTION
**Pull request by bhavanibandla**
### Jira link

https://transformuk.atlassian.net/browse/HOTT-2002

### What?

We've recently added a guide UI https://github.com/trade-tariff/trade-tariff-frontend/pull/1063 to show guidance to the user when a certain threshold of headings are associated with the same guide. This is part of beta search which is only enabled on development so we need to enable the exclusion of these tests in any test runs against staging or production (the majority of the changes in this PR make this happen). 

- [x] Added a new regression test file for beta search with cases for the guidance UI (see https://github.com/trade-tariff/trade-tariff-frontend/pull/1063)
- [x] Renamed all spec files to match cypress preferred internal spec pattern including .cy
- [x] Adjusted the ordering of the config loading to make sure anything in the .env file is included in the grep pattern
- [x] Removed a whole bunch of dead feature files which never actually get run

As an aside, we updated vim-test to reflect the internal change in cypress configuration (https://github.com/vim-test/vim-test/pull/697).

### Why

- This is required to test the guide UI and to enable devOnly tests not to run in staging/production test workflows

### AC

- Ran the production/staging regression suites to validate the change excludes devOnly tag
- Circle integration passes

### CODE REVIEW

File: cypress/e2e/HOTT-Shared/devSmokeTestCI.spec.js [hunk 1]
```diff
@@ -160,4 +161,30 @@ describe('🚀  UK 🇬🇧 XI 🇪🇺 💡 | devSmokeTestCI- UK,XI | Smoke tes
     cy.searchForCommodity('3808941000');
     cy.contains(/Commodity .*3808941000/i);
   });
+
+  // Search strategic - implement guides on the UI - HOTT-2002
+  ['uk', 'xi'].forEach(function(country) {
```
Comment (willfish) [inline cypress/e2e/HOTT-Shared/devSmokeTestCI.spec.js:166]: I'd probably not add this to the smoke tests that run on every dev build.

Are you able to add a separate regression file that will run overnight?

Comment (bhavanibandla) [reply cypress/e2e/HOTT-Shared/devSmokeTestCI.spec.js:166]: Can you confirm - the regression file for the Development environment?

Comment (willfish) [reply cypress/e2e/HOTT-Shared/devSmokeTestCI.spec.js:166]: So we have:

- Smoke tests run as part of a developers workflow when we deploy a new feature to development
- Smoke tests run as part of a deployment of a new feature to staging
- Regression tests (really just e2e tests) which run overnight for staging and production

I'm basically saying:

1. Do not add additional tests to the smoketests that aren't critical - we need a balance which integrates services but doesn't check every low-value feature
2. Add tagging such that tests run against staging and production regression workflows can exclude regression specs marked as for development, only
3. Enable tagging on a new regression file for beta search (what you're testing) that includes the development tag

Comment (willfish) [reply cypress/e2e/HOTT-Shared/devSmokeTestCI.spec.js:166]: Happy to pair on this, BTW

**Review by willfish — APPROVED** [review]

**Pull request merged**